draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1217,14 +1282,14 @@ Parameters:
 
 Inputs:
 - msg is the message to hash.
-- ctr is either 0 or 1.
+- ctr is either 0, 1, or 2.
   This is used to efficiently create independent
   instances of hash_to_base (see discussion above).
 
 Output: u, an element in F.
 
 Steps:
-1. m' = H(msg) || I2OSP(ctr, 1)
+1. m' = "HASH-TO-CURVE" || H(msg) || I2OSP(ctr, 1)
```

Comment:
Would we consider making HASH-TO-CURVE shorter (e..g "H2C"/ "HTC"), and potentially truncating H(msg) such that we can fit m' into a single block in the loop below?

E.g. the input to the hash in the loop becomes `"H2C" || H(msg)[..len(H) - 6] || I2OSP(ctr, 1)  || I2OSP(i, 1) || I2OSP(j, 1)`?

Reply:
Perhaps in a separate issue?

Reply:
we'd have to truncate a reasonable amount to avoid going past one block, because SHA-256 adds at least 65 bits of padding to the end of the message. Specifically, the padding is

- append a single '1' bit
- append k '0' bits where 0 <= k < 512 and bitlen(msg) + 1 + k + 64 is divisible by 512
- append a 64-bit representation of bitlen(msg)

so it would have to be something like H(msg)[..len(H) - 15], which is pretty extreme.

Also, we might want to resolve #137 first, since that would moot this discussion.